DataStructures/disjoint.h

```diff
@@ -0,0 +1,20 @@
+#ifndef DISJOINT_H
+#define DISJOINT_H
+
+#include <cstdlib>
+#include <vector>
+
+class disjoint_set_t
+{
+public:
+    disjoint_set_t(size_t nodes);
+
+    int find_set(int x);
```

Comment:
`find_set` is kind of a "misleading" name for this function, I think. Here are a few suggestions:
- `root`
- `parent`
- `get_root` or `find_root`
- `get_parent` or `find_parent`